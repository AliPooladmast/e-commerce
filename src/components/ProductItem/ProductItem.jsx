import React from "react";

const ProductItem = () => {
  return <div>ProductItem</div>;
};

export default ProductItem;
