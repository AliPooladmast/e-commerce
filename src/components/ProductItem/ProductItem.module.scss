.Container {
  flex: 1;
  margin: 5px;
  height: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #f5fdfb;
}

.ImgContainer {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  img {
    height: 250px;
    width: 250px;
    object-fit: cover;
    border-radius: 50%;
    z-index: 2;
  }
}
