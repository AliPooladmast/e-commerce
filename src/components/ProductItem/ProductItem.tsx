import { ShoppingCartOutlined, SearchOutlined } from "@mui/icons-material";
import React from "react";
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { addProduct } from "../../redux/cartSlice";
import style from "./ProductItem.module.scss";

interface IItem {
  _id: string;
  quantity: number;
  price: number;
  size: string;
  color: string;
  title: string;
  img?: string;
}

const ProductItem = ({ item }: { item: IItem }) => {
  const dispatch = useDispatch();

  const handleAddClick = () => {
    dispatch(
      addProduct({
        ...item,
        quantity: 1,
        size: item.size[0],
        color: item.color[0],
      })
    );
  };

  return (
    <div className={style.Wrapper}>
      <div className={style.Container}>
        <div className={style.ImgContainer}>
          <img src={item.img} alt="" />
        </div>
        <div className={style.Info}>
          <div onClick={handleAddClick}>
            <ShoppingCartOutlined />
          </div>
          <div>
            <Link to={`/product/${item._id}`} className={style.Link}>
              <SearchOutlined />
            </Link>
          </div>
        </div>
      </div>
      <div className={style.Title}>
        <div>{item.title}</div>
        <div className={style.Price}>${item.price}</div>
      </div>
    </div>
  );
};

export default ProductItem;
