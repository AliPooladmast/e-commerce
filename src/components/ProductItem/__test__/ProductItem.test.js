const { render, screen } = require("@testing-library/react");
const { BrowserRouter } = require("react-router-dom");
import ProductItem from "../ProductItem";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

const item = {
  title: "dress",
  size: "M",
  color: "blue",
  price: "10",
  _id: "123",
  img: "image",
};

it("match snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <ProductItem item={item} />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

describe("test item title and price", () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <ProductItem item={item} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("check the title", () => {
    expect(screen.getByTestId("item-title").textContent).toBe(item.title);
  });

  it("check the price", () => {
    expect(screen.getByTestId("item-price").textContent).toBe("$" + item.price);
  });

  it("check the product page redirect", () => {
    expect(
      screen.getByTestId("item-page-link").href.includes(`/product/${item._id}`)
    ).toBeTruthy();
  });
});
