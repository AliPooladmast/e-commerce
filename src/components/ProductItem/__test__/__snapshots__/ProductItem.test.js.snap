// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Wrapper"
>
  <div
    className="Container"
  >
    <div
      className="ImgContainer"
    >
      <img
        alt=""
        src="image"
      />
    </div>
    <div
      className="Info"
    >
      <div
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ShoppingCartOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
      <div>
        <a
          className="Link"
          data-testid="item-page-link"
          href="/product/123"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
  <div
    className="Title"
  >
    <div
      data-testid="item-title"
    >
      dress
    </div>
    <div
      className="Price"
      data-testid="item-price"
    >
      $
      10
    </div>
  </div>
</div>
`;
